src/viking.js

```diff
@@ -1,15 +1,101 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength){
+    this.health = health;
+    this.strength = strength;
+  }
+
+  attack(){
+   return this.strength
+  }
+
+  receiveDamage(damage){
+    this.damage = damage
+    this.health -= damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength){
+    super(health, strength)
+    this.name = name
+  }
+  
+  receiveDamage(damage){
+    this.damage = damage
+    this.health = this.health - damage;
+    return (this.health > 0) ? `${this.name} has received ${this.damage} points of damage` : `${this.name} has died in act of combat`;
+    } 
+    battleCry(){
+      return `Odin Owns You All!`
+    }
+    attack(){
+      return super.attack()
+     }
+  }
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  receiveDamage(damage){
+    this.damage = damage;
+    this.health = this.health - damage;
+    return (this.health > 0) ? `A Saxon has received ${this.damage} points of damage` : `A Saxon has died in combat`
+    }
+    attack(){
+      return super.attack()
+     }
+  }
 
 // War
-class War {}
+class War {
+  vikingArmy = [];
+  saxonArmy = [];
+
+  addViking(viking){
+    this.vikingArmy.push(viking)
+  }
+  
+  addSaxon(saxon){
+    this.saxonArmy.push(saxon)
+  }
+  vikingAttack(){
```

Comment:
Está bien cómo lo hiciste. Te pongo otra solución con un filter para que tengas otra visión!! 
`  vikingAttack() {
    const viking = randomElement(this.vikingArmy);
    const saxon = randomElement(this.saxonArmy);

    const attackResult = saxon.receiveDamage(viking.strength);
    this.saxonArmy = this.saxonArmy.filter(saxon => saxon.health > 0);
    return attackResult;
  }`